python/pyspark/sql/udtf.py

```diff
@@ -61,6 +61,26 @@ class AnalyzeArgument:
     is_table: bool
 
 
+@dataclass(frozen=True)
+class PartitioningColumn:
+    """
+    Represents a UDTF column for purposes of returning metadata from the 'analyze' method.
+    """
+
+    name: str
+
+
+@dataclass(frozen=True)
+class OrderingColumn:
```

Comment:
SortOrder class also includes `nullOrdering` . Do we want to add is here? For example `ORDER BY t.c1 DESC NULL FIRST`